synapse/http/server.py

```diff
@@ -275,6 +275,10 @@ class DirectServeJsonResource(_AsyncResource):
     formatting responses and errors as JSON.
     """
 
+    def __init__(self, canonical_json=False, extract_context=False):
+        super().__init__(extract_context)
+        self.canonical_json = canonical_json
+
```

Comment:
this change means that the first positional param of the constructor is now `canonical_json`, whereas it was previously `extract_context`. I thought that keeping the consistency with `JsonResource` was the lesser evil. I did go through the codebase to check that nothing other than `JsonResource` was making a call which would be misinterpreted.